src/backend/utilities/initKilt.ts

```diff
@@ -1,8 +1,33 @@
 import { init } from '@kiltprotocol/core';
+import { BlockchainApiConnection } from '@kiltprotocol/chain-helpers';
 
 import { configuration } from './configuration';
+import { logger } from './logger';
+import { trackConnectionState } from './trackConnectionState';
 
-export async function initKilt(): Promise<true> {
+export async function initKilt(): Promise<void> {
   await init({ address: configuration.blockchainEndpoint });
-  return true;
+}
+
+export const blockchainConnectionState = trackConnectionState(60 * 1000);
+
+export async function disconnectHandler(value?: string) {
+  blockchainConnectionState.off();
+  logger.warn(value, 'Received disconnect even from the blockchain');
+
+  while (true) {
+    try {
+      await reConnect();
```

Comment:
Is this confirmed to be working? AFAICS, the `getConnectionOrConnect();` function would just return the internally cached instance and I am not sure what happens if you `await` that object after it was already connected before.

Reply:
Maybe it makes sense to use the `BlockchainApiConnection.connected()` function (which should be renamed to `isConnected`), because the api already tries to reconnect automatically?

Reply:
No, we haven’t tested that. We rely on your expertise in the SDK here. The `connected` doesn’t wait for reconnection though. Should I instead use the "connected" event on the existing object?

Reply:
yes, listening for the connected event is a good idea

Reply:
Is it better now?